From 70cd86a5eef103137e64f2e3868a5bd2d58d71d7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Gregor=20T=C3=A4tzner?= <gregor@freenet.de>
Date: Thu, 26 Jul 2012 14:32:38 +0200
Subject: [PATCH] Fix deletion of preferred application launchers

In default panel config the host of that magic preferred app url is used
as the key. So lets use that key to delete the launchers, too.

BUG: 278724
FIXED-IN: 4.9.1
REVIEW: 105374
---
 libs/taskmanager/groupmanager.cpp |   14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/libs/taskmanager/groupmanager.cpp b/libs/taskmanager/groupmanager.cpp
index 5ca0159..45c15a9 100644
--- a/libs/taskmanager/groupmanager.cpp
+++ b/libs/taskmanager/groupmanager.cpp
@@ -1070,8 +1070,16 @@ void GroupManagerPrivate::unsaveLauncher(LauncherItem *launcher)
         return;
     }
 
-    if (cg.hasKey(launcher->name())) {
-        cg.deleteEntry(launcher->name());
+    QString launcherKey;
+
+    if (launcher->launcherUrl().protocol() == "preferred")
+        // in default config the host of the preferred application url is used as key
+        launcherKey = launcher->launcherUrl().host();
+    else
+        launcherKey = launcher->name();
+
+    if (cg.hasKey(launcherKey)) {
+        cg.deleteEntry(launcherKey);
         emit q->configChanged();
     }
 }
@@ -1127,7 +1135,7 @@ int GroupManagerPrivate::launcherIndex(const KUrl &url)
     foreach (const LauncherItem * item, launchers) {
         if (item->launcherUrl() == url) {
             return index;
-        } 
+        }
 
         ++index;
     }
-- 
1.7.10.4

